src/main/resources/jenkins/plugins/openstack/compute/OpenStack.groovy

```diff
@@ -0,0 +1,89 @@
+package jenkins.plugins.openstack.compute
+
+import jenkins.plugins.openstack.compute.internal.DestroyMachine
+import jenkins.plugins.openstack.compute.internal.Openstack
+import org.openstack4j.model.compute.Server
+
+class OpenStack implements Serializable {
+
+    private org.jenkinsci.plugins.workflow.cps.CpsScript script
+
+    public OpenStack(org.jenkinsci.plugins.workflow.cps.CpsScript script) {
+            this.script = script
+    }
+
+    public Node server(String cloudName, String template) {
+        new Node(this, cloudName, template)
+    }
+
+    public static class Node implements Serializable {
+
+        private final OpenStack openstack
+        public final String cloudName
+        public final String template
+        private Server server = null
+
+        private Node(OpenStack openstack, String cloudName, String template) {
+            this.openstack = openstack
+            this.cloudName = cloudName
+            this.template = template
+        }
+
+        public void destroy() {
+            if (server != null) {
+                DestroyMachine dm
+                dm = new DestroyMachine(cloudName, server.getId())
+                dm.dispose()
+                server = null
+            } else {
+                //Cant destroy what isn't created
```

Comment:
I prefer moving the boot semantics to the global object level so users will call `openstack.boot("MyCloud", "testDb", SCOPE)` (or `openstack.cloud("MyCloud").template("testDb").boot(SCOPE)`) so it would guarantee that every `Node` instance have exactly one machine behind it. With current design there is a risk someone boots multiple machines using one `Node` instance and it will be able to destroy only the last one.